maps/southern_cross/events/wildlife_encounter.dm

```diff
@@ -0,0 +1,95 @@
+// This event sends random sif mobs at someone in the wilderness, unannounced.
+
+/datum/event2/meta/wildlife_encounter
+	name = "random encounter"
+	departments = list(DEPARTMENT_EVERYONE)
+	chaos = 10
+	event_type = /datum/event2/event/wildlife_encounter
+
+/datum/event2/meta/wildlife_encounter/get_weight()
+	var/explorers = metric.count_people_with_job(/datum/job/explorer) + metric.count_people_with_job(/datum/job/sar)
+
+	if(!explorers)
+		return 0
+	return 20 + explorers * 50
+
+/datum/event2/event/wildlife_encounter
+	var/mob/living/victim = null
+
+/datum/event2/event/wildlife_encounter/set_up()
+	var/list/potential_victims = list()
+
+	for(var/mob/living/L in player_list)
+		//if(!(L.z in get_location_z_levels()))
+		//	log_debug("Not on the right z-level")
+		//	continue // Not on the right z-level.
+
+		if(L.stat)
+			continue // Don't want dead people.
+		if(istype(get_turf(L), /turf/simulated/floor/outdoors) && istype(get_area(L),/area/surface/outside/wilderness))
+			potential_victims += L
+
+	if(potential_victims.len)
+		victim = pick(potential_victims)
+		potential_victims = null
+
+/datum/event2/event/wildlife_encounter/start()
+	if(!victim)
+		log_debug("Failed to find a target for random encounter. Aborting.")
+		abort()
+		return
+
+	var/number_of_packs = rand(1, 3)
+	// Getting off screen tiles is kind of tricky due to potential edge cases that could arise.
+	// The method we're gonna do is make a big square around the victim, then
+	// subtract a smaller square in the middle for the default vision range.
+
+	var/list/outer_square = get_safe_square(victim, world.view + 5)
+	var/list/inner_square = get_safe_square(victim, world.view + 1)
```

Comment:
Given that this doesn't consider the positions of any other players, it's _possible_ that this will spawn mobs within the view of another player. However, I'm not sure that trying to consider the views of other players is necessarily feasible, since you could have a large number of players to consider (Even if you track those players that could interfere from the victim selection), which could altogether disqualify a particular victim because nowhere in the predefined donut is not visible by _any_ player.

So I'm mostly just noting that this behaviour may be observed by players, and (especially without any announcement), it could be considered buggy by such players, particularly if/when a hostile mob like a phoron spider happens, through random chance, to spawn right next to someone.

Reply:
It now checks if anyone else on the victim's list is within 7 tiles of the spawn turf, so it shouldn't be possible to see unless you're using binoculars, and looking in exactly the direction it decides to spawn.